File: C/INICIANTE/1181 Linha na Matriz.c
#include <stdio.h>
int main()
{
    int linhaDaOperacao, i, j;
    char operacao[2];
    double matriz[12][12], soma;
    soma = 0;
    scanf("%d %s", &linhaDaOperacao, operacao);
    for (i=0; i<12; i++)
        for (j=0; j<12; j++)
            scanf("%lf", &matriz[i][j]);
    for (i=0; i<12; i++)
        for (j=0; j<12; j++)
            if (i == linhaDaOperacao)
                soma += matriz[i][j];
    if (operacao[0] == 'S')
        printf("%.1lf\n", soma);
    else
        printf("%.1lf\n", (soma/12.0));
    return 0;
}

File: Python/BEGINNER/1181 Linha na Matriz.py
# SUBPROGRAMA
def fazerMatriz():
	minhaMatriz = []
	for col in range(12):
		linha = []
		for lin in range(12):
			linha.append(float(input()))
		minhaMatriz.append(linha)
	return minhaMatriz


def somarLinha(linhaOp, matriz):
	somaDaLinha = 0
	for col in range(len(matriz)):
		somaDaLinha += matriz[linhaOp][col]
	return somaDaLinha


def mediarLinha(linhaOp, matriz):
	somaLinha = somarLinha(linhaOp, matriz)
	mediaDaLinha = somaLinha / len(matriz[linhaOp])
	return mediaDaLinha


# PRINCIPAL
linhaDaOperacao = int(input())
operacao = input()
essaMatriz = fazerMatriz()
if operacao == 'S':
	soma = somarLinha(linhaDaOperacao, essaMatriz)
	print('%.1f' % soma)
else:
	media = mediarLinha(linhaDaOperacao, essaMatriz)
	print('%.1f' % media)
